Sources/hostmgr-helper/VirtualMachineSlot.swift

```diff
@@ -17,144 +17,237 @@ class VirtualMachineSlot: NSObject, ObservableObject {
-            }
-            self.managedVirtualMachine = nil
+    /// - This method does not clean up stopped VMs.
+    /// - This method throws an error if trying to stop in a state other than `.starting` and `.running`.
+    /// - Parameters:
+    ///   - error: Error to include if the slot is stopping because of an error.
+    func stop(withError error: Error? = nil) async throws {
+        Logger.helper.log("Stopping \(role.displayName) slot with current status \(state).")
+        switch state {
+        case .starting(let config, let task):
+            // MARK: Starting -> Stopping
+            self.state = .stopping(config)
+            task.cancel()
+        case .running(let mvm):
+            // MARK: Running -> Stopping
+            self.state = .stopping(mvm.config)
+            await stopVm(mvm.machine, handle: mvm.handle)
+        default:
+            Logger.helper.error("Stop called while state was \(state).")
+            throw Errors.invalidStopState
         }
 
         if let error {
+            // MARK: Stopping -> Crashed
             Logger.helper.error("Resetting slot with crashed state: \(error)")
-            self.status = .crashed(error)
+            self.state = .crashed(error)
         } else {
-            self.status = .empty
+            // MARK: Stopping -> Empty
+            Logger.helper.log("Resetting slot to empty state.")
+            self.state = .empty
         }
     }
 
-    @MainActor
     func isConfiguredForHandle(_ handle: String) -> Bool {
-        guard let configHandle = managedVirtualMachine?.config.handle else {
+        guard let slotHandle = state.handle else {
+            Logger.helper.debug(
+                "\(role.displayName) slot had no handle configured."
+            )
             return false
         }
+
         Logger.helper.debug(
-            "Comparing \(configHandle) and \(handle)"
+            "Comparing \(slotHandle) and \(handle)."
         )
-        return configHandle == handle
+        return slotHandle == handle
     }
 
-    @MainActor
     var isAvailable: Bool {
-        switch self.status {
+        Logger.helper.debug(
+            "Slot availability check: \(role.displayName) slot has \(state) status."
+        )
+
+        switch self.state {
         case .empty, .crashed: return true
         default: return false
         }
     }
+
+    /// Creates a running VM inside a `ManagedVirtualMachine` configured by a `LaunchConfiguration`.
+    ///
+    /// If any errors occur after the VM was started, this method will attempt to stop the VM and perform a cleanup.
+    /// - Parameter launchConfiguration: VM configuration.
+    /// - Returns: A `ManagedVirtualMachine` containing a running VM.
+    private func createManagedVm(launchConfiguration: LaunchConfiguration) async throws -> ManagedVirtualMachine {
+        Logger.helper.log("Creating VM \(launchConfiguration.handle).")
+        let virtualMachine = try await launchConfiguration.setupVirtualMachine()
```

Comment:
I'll wait for the first review pass, but I realize a small tweak needs to be made here where even a newly created VM should get cleaned up in this method.

Reply:
Those changes landed in 600a776928af79e0cc350ce265eb8f08a5e9f15f.